### PR TITLE
generate documentation

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -126,6 +127,12 @@ module.exports = {
 
         // Type conversions require an explicit cast in AssemblyScript.
         '@typescript-eslint/no-unnecessary-type-assertion': 'off',
+
+        // Disable jsdoc
+        'require-jsdoc': 'off',
+        'valid-jsdoc': 'off',
+        // Enable typedoc
+        "tsdoc/syntax": "error",
```
Comment (gregLibert) [inline .eslintrc.cjs:135]: single quote, double => make a choice and be consistent :laughing:

File: assembly/collections/persistentMap.ts [hunk 2]
```diff
@@ -43,16 +43,16 @@ export class PersistentMap<K, V> {
...
    */
   constructor(prefix: string) {
     this._elementPrefix = prefix + _KEY_ELEMENT_SUFFIX;
     this._size = 0;
   }
 
   /**
-   * @param {string} key - Search key.
-   * @return {string} An internal string key for a given key of type K.
+   * @param key - Search key.
```
Comment (gregLibert) [inline assembly/collections/persistentMap.ts:54]: why mixing @param and @typeParam ?

File: assembly/std/context.ts [hunk 3]
```diff
@@ -71,7 +67,7 @@ export function addressStack(): Array<Address> {
  * Caller is the person or the smart contract that directly called
  * the pending function.
  *
- * @return {Address} returns the stack element just below the top of the address stack
+ * @returns the stack element just below the top of the address stack
```
Comment (gregLibert) [inline assembly/std/context.ts:70]: Man, that comment about the returned value is scary. You should either delete it or just say it returns the caller's address.

File: assembly/std/context.ts [hunk 4]
```diff
@@ -83,7 +79,7 @@ export function caller(): Address {
  *
  * Callee is the current smart-contract address.
  *
- * @return {Address} returns the stack element at the top of the address stack
+ * @returns the stack element at the top of the address stack
```
Comment (gregLibert) [inline assembly/std/context.ts:82]: same. the callee concept is maybe unclear. You can precise it by adding a synonym such as current smart contract address ?

File: assembly/collections/persistentMap.ts [hunk 5]
```diff
@@ -70,22 +70,21 @@ export class PersistentMap<K, V> {
...
+   * @returns True if the given key present in the map.
    */
   contains(key: K): bool {
     return Storage.has(this._key(key));
   }
 
   /**
    * Returns the map size
-   *
    * ```ts
```
Comment (Thykof) [inline assembly/collections/persistentMap.ts:82]:
```suggestion
   *
   * @example
   * ```ts
```

**Review by gregLibert — COMMENTED** [review]
Requested changes.

We really need to update the documentation. Especially on the comments of the returned values...

File: .eslintrc.cjs [hunk 6]
```diff
@@ -1,8 +1,11 @@
 module.exports = {
   root: true,
   parser: '@typescript-eslint/parser',
-  plugins: ['@typescript-eslint'],
-  ignorePatterns: ['assembly/abi.ts'],
+  plugins: [
+    "@typescript-eslint/eslint-plugin",
+    "eslint-plugin-tsdoc"
```
Comment (Thykof) [inline .eslintrc.cjs:6]:
```suggestion
    '@typescript-eslint/eslint-plugin',
    'eslint-plugin-tsdoc'
```

File: .eslintrc.cjs (hunk 1 above)
Comment (Thykof) [inline .eslintrc.cjs:135]:
```suggestion
        'tsdoc/syntax': 'error',
```

Comment (Thykof) [reply .eslintrc.cjs:135]: good catch! :)

File: assembly/collections/persistentMap.ts (hunk 2 above)
Comment (Thykof) [reply assembly/collections/persistentMap.ts:54]: typeParam is for generic type https://typedoc.org/tags/typeParam/

File: assembly/std/context.ts (hunk 3 above)
Comment (Thykof) [inline assembly/std/context.ts]:
```suggestion
 * @returns the caller's address
```

File: assembly/std/context.ts (hunk 4 above)
Comment (Thykof) [inline assembly/std/context.ts]:
```suggestion
 * @returns the called smart contract address
```

**Review by Thykof — APPROVED** [review]

File: assembly/std/context.ts [hunk 7]
```diff
@@ -71,7 +67,7 @@ export function addressStack(): Array<Address> {
  * Caller is the person or the smart contract that directly called
  * the pending function.
  *
- * @return {Address} returns the stack element just below the top of the address stack
+ * @returns the caller address
```
Comment (Thykof) [inline assembly/std/context.ts:70]:
```suggestion
 * @returns the caller's address
```

File: assembly/std/context.ts [hunk 8]
```diff
@@ -83,7 +79,7 @@ export function caller(): Address {
  *
  * Callee is the current smart-contract address.
  *
- * @return {Address} returns the stack element at the top of the address stack
+ * @returns the callee address
```
Comment (Thykof) [inline assembly/std/context.ts:82]:
```suggestion
 * @returns the called smart contract address
```

File: assembly/collections/persistentMap.ts (hunk 2 above)
Comment (gregLibert) [reply assembly/collections/persistentMap.ts:54]: The @typeParam tag is used to document a type parameter of a function, method, class, interface or type alias.

**Pull request merged**